src/logworker.cpp

```diff
@@ -120,6 +120,20 @@ namespace g3 {
       return std::unique_ptr<LogWorker>(new LogWorker);
    }
 
+   void LogWorker::removeWrappedSink(std::shared_ptr<g3::internal::SinkWrapper> sink) {
+      auto bg_removesink_call = [this, sink] {
+         for (auto it = _impl._sinks.begin(); it != _impl._sinks.end(); ) {
+            if ((*it) == sink) {
+               it = _impl._sinks.erase(it);
```

Comment:
You can use the std::find  also.

auto& v = _impl._sinks;
auto it = std::find(v.begin(), v.end(), sink);
if(it != v.end())
    v.erase(it);